modules/standard/CTypes.chpl

```diff
@@ -954,6 +956,25 @@ module CTypes {
     }
     return c_pointer_return(b.buff[0]);
   }
+
+  @chpldoc.nodoc
+  inline proc c_ptrTo_helper(ref c: class): c_ptr(void)
+  {
+    if (isUnmanagedClass(c)) {
+      return __primitive("cast", c_ptr(void), c);
+    } else {
+      return __primitive("cast", c_ptr(void), c.borrow());
+    }
+  }
+  @chpldoc.nodoc
+  inline proc c_ptrTo_helper(ref c: class?): c_ptr(void)
+  {
+    if (isUnmanagedClass(c)) {
+      return __primitive("cast", c_ptr(void), c);
+    } else {
+      return __primitive("cast", c_ptr(void), c.borrow());
+    }
+  }
```

Comment:
If I understand right, the helper is needed for internal usage (like in IO) since `c_ptrTo` requires a compile time flag? But since the definition is the same as `c_ptrTo`, can you use the helper to implement the functions?

Reply:
Yep that's what it's for, and will be removed once we get rid of the old behavior (and the flag) entirely. However I lean against calling the helper in the main function because:

- It's a single duplicated line (after removing the logic branch for unmanaged)
- The helper will be removed soon and then need to be removed from the main function
- Duplication is what the other `c_ptrTo_helper` overloads do

Reply:
Works for me